tests/unit/utils/getUserDetails.test.ts

```diff
@@ -0,0 +1,53 @@
+import { getUserDetails } from "../../../src/utils/getUserDetails";
+import { FAILED_TO_FETCH_TASKS } from "../../../src/constants/responses";
+import { RDS_BASE_API_URL } from "../../../src/constants/urls";
+import JSONResponse from "../../../src/utils/JsonResponse";
+
+describe("Test getUserDetails function", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Should return user details", async () => {
+    const userData = {
+      user: {
+        discordId: "123456",
+        first_name: "John",
+        last_name: "Doe",
+        discordJoinedAt: "2023-09-13T09:30:00.000Z",
+        state: "Active",
+      },
+    };
+
+    jest
+      .spyOn(global, "fetch")
+      .mockImplementation(() => Promise.resolve(new JSONResponse(userData)));
+
+    const userId = "123456";
+    const result = await getUserDetails(userId);
```

Comment:
Nit: you are taking userId or discordId as a params

Reply:
Eventhough it is userId here, but internally it will take discordId.